src/generators.cpp

```diff
@@ -179,18 +171,21 @@ std::string to_string(DeviceType device_type) {
       return "WebGpu";
     case DeviceType::QNN:
       return "QnnWithSharedMemory";
+    default:
+      throw std::runtime_error("Unknown device type");
   }
-  throw std::runtime_error("Unknown device type");
 }
 
 DeviceInterface* GetDeviceInterface(DeviceType type) {
   switch (type) {
     default:
     case DeviceType::CPU:
       return GetCpuInterface();
-#if USE_CUDA
     case DeviceType::CUDA:
       return GetCudaInterface();
+#if USE_DML
```

Comment:
Why do we need this #if and not one for CUDA?

Reply:
It's not built as a shared library yet, only cuda is. So we can't just try loading it and failing if it doesn't exist. It's either built with it or not. Once it's a shared library there will be no #ifdef.

I could have the function exist but the definition of this function will be inside another #ifdef !USE_DML so it's just moving the problem around.